### ISSUE TITLE
[Bug]: [benchmark][cluster] Collection has been released during rolling upgrade image

### ISSUE THREAD

**Issue by wangting0128** [milvus-io/milvus#26679]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: v2.3.0
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka): pulsar   
- SDK version(e.g. pymilvus v2.0.0rc2): 2.3.0
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

argo task: fouramf-7bnvj
upgrade argo task: fouramf-4gt6w

server:
```
[2023-08-29 08:13:31,694 -  INFO - fouram]: 
NAME                                                      STATUS      RESTARTS     AGE       IP                NODE           
lb-rolling-upgrade-milvus-datacoord-b87f769c7-vfdgp       Running     0            1h14m     10.104.23.156     4am-node27     
lb-rolling-upgrade-milvus-datanode-64dd74c458-gzbjg       Running     0            1h14m     10.104.14.182     4am-node18     
lb-rolling-upgrade-milvus-indexcoord-66f9484cd5-tkg2m     Running     0            1h14m     10.104.1.204      4am-node10     
lb-rolling-upgrade-milvus-indexnode-66f95d78d4-25jc6      Running     0            1h14m     10.104.6.6        4am-node13     
lb-rolling-upgrade-milvus-indexnode-66f95d78d4-d5g7g      Running     0            1h14m     10.104.13.48      4am-node16     
lb-rolling-upgrade-milvus-proxy-78f79f7856-hd8fq          Running     0            1h14m     10.104.1.205      4am-node10     
lb-rolling-upgrade-milvus-querycoord-7f4c8dd545-btw4q     Running     0            1h14m     10.104.1.206      4am-node10     
lb-rolling-upgrade-milvus-querynode-596d676944-757s4      Running     0            1h14m     10.104.9.119      4am-node14     
lb-rolling-upgrade-milvus-querynode-596d676944-bcbvm      Running     0            1h14m     10.104.17.253     4am-node23     
lb-rolling-upgrade-milvus-querynode-596d676944-z8n7k      Running     0            1h14m     10.104.23.154     4am-node27     
lb-rolling-upgrade-milvus-rootcoord-5664f85457-hhtn5      Running     0            1h14m     10.104.23.155     4am-node27     
lb-rolling-upgrade-etcd-0                                 Running     1            1h20m     10.104.24.55      4am-node29     
lb-rolling-upgrade-etcd-1                                 Running     1            1h20m     10.104.21.196     4am-node24     
lb-rolling-upgrade-etcd-2                                 Running     0            1h20m     10.104.14.180     4am-node18     
lb-rolling-upgrade-pulsar-bookie-0                        Running     0            1h20m     10.104.24.71      4am-node29     
lb-rolling-upgrade-pulsar-bookie-1                        Running     0            1h20m     10.104.21.200     4am-node24     
lb-rolling-upgrade-pulsar-bookie-2                        Running     0            1h20m     10.104.14.179     4am-node18     
lb-rolling-upgrade-pulsar-broker-0                        Running     0            1h20m     10.104.14.173     4am-node18     
lb-rolling-upgrade-pulsar-proxy-0                         Running     0            1h20m     10.104.21.194     4am-node24     
lb-rolling-upgrade-pulsar-recovery-0                      Running     0            1h20m     10.104.21.193     4am-node24     
lb-rolling-upgrade-pulsar-zookeeper-0                     Running     0            1h20m     10.104.24.68      4am-node29     
lb-rolling-upgrade-pulsar-zookeeper-1                     Running     0            1h19m     10.104.19.112     4am-node28     
lb-rolling-upgrade-pulsar-zookeeper-2                     Running     0            1h17m     10.104.23.148     4am-node27     
lb-rolling-upgrade-minio-0                                Running     0            1h20m     10.104.21.197     4am-node24     
lb-rolling-upgrade-minio-1                                Running     0            1h20m     10.104.14.178     4am-node18     
lb-rolling-upgrade-minio-2                                Running     0            1h20m     10.104.24.60      4am-node29     
lb-rolling-upgrade-minio-3                                Running     0            1h20m     10.104.9.97       4am-node14     
 (common_func.py:407)
[2023-08-29 08:13:31,695 -  INFO - fouram]: [Base] upgrade configs: {'spec': {'components': {'enableRollingUpdate': True, 'imageUpdateMode': 'rollingUpgrade'}, 'mode': 'cluster'}, 'apiVersion': 'milvus.io/v1beta1', 'kind': 'Milvus'} (base.py:194)
[2023-08-29 08:13:31,952 -  INFO - fouram]: [wait_for_healthy] Waiting for instance:lb-rolling-upgrade health... (operator_client.py:192)
[2023-08-29 08:14:02,099 -  INFO - fouram]: [wait_for_healthy] Waiting for instance:lb-rolling-upgrade health... (operator_client.py:192)
[2023-08-29 08:14:32,281 -  INFO - fouram]: [wait_for_healthy] Waiting for instance:lb-rolling-upgrade health... (operator_client.py:192)
[2023-08-29 08:15:02,452 -  INFO - fouram]: [wait_for_healthy] Waiting for instance:lb-rolling-upgrade health... (operator_client.py:192)
[2023-08-29 08:15:32,759 -  INFO - fouram]: [wait_for_healthy] Waiting for instance:lb-rolling-upgrade health... (operator_client.py:192)
[2023-08-29 08:16:02,917 -  INFO - fouram]: [wait_for_healthy] Waiting for instance:lb-rolling-upgrade health... (operator_client.py:192)
[2023-08-29 08:16:33,076 -  INFO - fouram]: [wait_for_healthy] Instance:lb-rolling-upgrade is healthy. (operator_client.py:189)
[2023-08-29 08:16:33,237 -  INFO - fouram]: [Base] Get pods after upgrade... (base.py:202)
[2023-08-29 08:16:34,030 -  INFO - fouram]: [get_pods] pod details of release(lb-rolling-upgrade):  (operator_client.py:307)
[2023-08-29 08:16:34,031 -  INFO - fouram]: 
NAME                                                      STATUS      RESTARTS     AGE       IP                NODE           
lb-rolling-upgrade-milvus-datacoord-5f94d9678f-mg6wl      Running     0            3m        10.104.19.200     4am-node28     
lb-rolling-upgrade-milvus-datanode-564787cd54-ssff9       Running     0            3m        10.104.21.35      4am-node24     
lb-rolling-upgrade-milvus-indexcoord-5bfb4d9769-r4s2z     Running     0            3m        10.104.19.197     4am-node28     
lb-rolling-upgrade-milvus-indexnode-7b766cc4d-4cvn4       Running     0            2m        10.104.19.201     4am-node28     
lb-rolling-upgrade-milvus-indexnode-7b766cc4d-tdmkt       Running     0            3m        10.104.12.224     4am-node17     
lb-rolling-upgrade-milvus-proxy-7998dd4568-wq27r          Running     0            3m        10.104.9.188      4am-node14     
lb-rolling-upgrade-milvus-querycoord-995954bf7-mwpbp      Running     0            3m        10.104.9.187      4am-node14     
lb-rolling-upgrade-milvus-querynode-5d89445b76-4hb6b      Running     0            2m        10.104.9.191      4am-node14     
lb-rolling-upgrade-milvus-querynode-5d89445b76-68snl      Running     0            1m        10.104.17.85      4am-node23     
lb-rolling-upgrade-milvus-querynode-5d89445b76-vpgq5      Running     0            3m        10.104.4.166      4am-node11     
lb-rolling-upgrade-milvus-rootcoord-784c66bb89-ntksw      Running     0            3m        10.104.19.199     4am-node28     
lb-rolling-upgrade-etcd-0                                 Running     1            1h23m     10.104.24.55      4am-node29     
lb-rolling-upgrade-etcd-1                                 Running     1            1h23m     10.104.21.196     4am-node24     
lb-rolling-upgrade-etcd-2                                 Running     0            1h23m     10.104.14.180     4am-node18     
lb-rolling-upgrade-pulsar-bookie-0                        Running     0            1h23m     10.104.24.71      4am-node29     
lb-rolling-upgrade-pulsar-bookie-1                        Running     0            1h23m     10.104.21.200     4am-node24     
lb-rolling-upgrade-pulsar-bookie-2                        Running     0            1h23m     10.104.14.179     4am-node18     
lb-rolling-upgrade-pulsar-broker-0                        Running     0            1h23m     10.104.14.173     4am-node18     
lb-rolling-upgrade-pulsar-proxy-0                         Running     0            1h23m     10.104.21.194     4am-node24     
lb-rolling-upgrade-pulsar-recovery-0                      Running     0            1h23m     10.104.21.193     4am-node24     
lb-rolling-upgrade-pulsar-zookeeper-0                     Running     0            1h23m     10.104.24.68      4am-node29     
lb-rolling-upgrade-pulsar-zookeeper-1                     Running     0            1h22m     10.104.19.112     4am-node28     
lb-rolling-upgrade-pulsar-zookeeper-2                     Running     0            1h20m     10.104.23.148     4am-node27     
lb-rolling-upgrade-minio-0                                Running     0            1h23m     10.104.21.197     4am-node24     
lb-rolling-upgrade-minio-1                                Running     0            1h23m     10.104.14.178     4am-node18     
lb-rolling-upgrade-minio-2                                Running     0            1h23m     10.104.24.60      4am-node29     
lb-rolling-upgrade-minio-3                                Running     0            1h23m     10.104.9.97       4am-node14 
```
![image](https://github.com/milvus-io/milvus/assets/26307815/8a52e596-faee-4941-a133-885442d92487)

client logs:
<img width="1431" alt="截屏2023-08-29 17 35 23" src="https://github.com/milvus-io/milvus/assets/26307815/435017c3-dacf-4811-a460-e2e9da742580">

### Steps To Reproduce

```markdown
1. deploy a Milvus Cluster
2. prepare 10m data and concurrent search 
3. rolling upgrade image from v2.3.0 to v2.3.0 during searching <- causing collection has been released
```

### Anything else?

test configs:
deploy_tool: op
deploy_config: fouramf-server-op-replicas-lb-3qn
other_params:
```
--milvus_tag=v2.3.0 --release_name=lb-rolling-upgrade --deploy_mode=cluster -k test_concurrent_locust_custom_parameters -s --case_params='{
     "load_params": {
          "replica_number": 3
     },
     "collection_params": {
          "collection_name": "lb_replica3",
          "shards_num": 1
     },
     "dataset_params": {
          "dim": 128,
          "dataset_name": "sift",
          "dataset_size": "10m",
          "ni_per": 50000,
          "metric_type": "L2"
     },
     "index_params": {
          "index_type": "HNSW",
          "index_param": {
               "M": 8,
               "efConstruction": 200
          }
     },
     "concurrent_params": {
          "concurrent_number": 300,
          "during_time": "3h",
          "interval": 20
     },
     "concurrent_tasks": [
          {
               "type": "search",
               "weight": 1,
               "params": {
                    "nq": 10,
                    "top_k": 10,
                    "search_param": {
                         "ef": 64
                    },
                    "timeout": 3600
               }
          }
     ]
}'
```

fouramf-server-op-replicas-lb-3qn:
```
    spec:
      components: 
        queryNode:
          replicas: 3
          resources:
            limits:
              cpu: '8'
              memory: 16Gi
            requests:
              cpu: '8'
              memory: 16Gi
        indexNode:
          resources:
            limits:
              cpu: '16.0'
              # memory: 4Gi
              memory: 8Gi
            requests:
              cpu: '5.0'
              # memory: 3Gi
              memory: 5Gi
          replicas: 2
        dataNode:
          resources:
            limits:
              cpu: '2.0'
              memory: 2Gi
            requests:
              cpu: '2.0'
              memory: 2Gi
```

**Comment by weiliu1031** [comment]
1. cause dc will register ic's session during start, but doesn't revoke ic's session during stop, which will block the new dc until session expired, which is about `60s` for now, should be fixed by #26680
2. during qc restart, if there is no available dc, and qc will failed to recover collection，then the collection has been released.
a better choice is to recover collection in async in new qc. recover collection won't block query coord's start

**Comment by wangting0128** [comment]
### Verification passed
image tag：master-20230830-a8e5dc35
argo task：fouramf-244dl

test result：
```
{'server': {'deploy_tool': 'operator',
            'deploy_mode': 'cluster',
            'config_name': 'cluster_8c16m',
            'config': {'spec': {'components': {'queryNode': {'resources': {'limits': {'cpu': '8',
                                                                                      'memory': '16Gi'},
                                                                           'requests': {'cpu': '8',
                                                                                        'memory': '16Gi'}},
                                                             'replicas': 3},
                                               'indexNode': {'resources': {'limits': {'cpu': '16.0',
                                                                                      'memory': '8Gi'},
                                                                           'requests': {'cpu': '5.0',
                                                                                        'memory': '5Gi'}},
                                                             'replicas': 2},
                                               'dataNode': {'resources': {'limits': {'cpu': '2.0',
                                                                                     'memory': '2Gi'},
                                                                          'requests': {'cpu': '2.0',
                                                                                       'memory': '2Gi'}}},
                                               'image': 'harbor.milvus.io/milvus/milvus:master-20230830-a8e5dc35'},
                                'mode': 'cluster',
                                'dependencies': {'etcd': {'inCluster': {'deletionPolicy': 'Delete',
                                                                        'pvcDeletion': True,
                                                                        'values': {'global': {'storageClass': 'local-path'},
                                                                                   'metrics': {'enabled': True,
                                                                                               'podMonitor': {'enabled': True}}}}},
                                                 'pulsar': {'inCluster': {'deletionPolicy': 'Delete',
                                                                          'pvcDeletion': True,
                                                                          'values': {'bookkeeper': {'volumes': {'journal': {'storageClassName': 'local-path'},
                                                                                                                'ledgers': {'storageClassName': 'local-path'}}},
                                                                                     'zookeeper': {'volumes': {'data': {'storageClassName': 'local-path'}}}}}},
                                                 'kafka': {'inCluster': {'deletionPolicy': 'Delete',
                                                                         'pvcDeletion': True,
                                                                         'values': {'persistence': {'storageClass': 'local-path'}}}},
                                                 'storage': {'inCluster': {'deletionPolicy': 'Delete',
                                                                           'pvcDeletion': True,
                                                                           'values': {'persistence': {'storageClass': 'local-path'},
                                                                                      'metrics': {'podMonitor': {'enabled': True}}}}}},
                                'config': {'log': {'level': 'debug'}}},
                       'apiVersion': 'milvus.io/v1beta1',
                       'kind': 'Milvus',
                       'metadata': {'name': 'fouram-op-94-9465'}},
            'host': 'lb-rolling-upgrade-milvus.qa-milvus',
            'port': '19530',
            'uri': ''},
 'client': {'test_case_type': 'ConcurrentClientBase',
            'test_case_name': 'test_concurrent_locust_custom_parameters',
            'test_case_params': {'dataset_params': {'metric_type': 'L2',
                                                    'dim': 128,
                                                    'dataset_name': 'sift',
                                                    'dataset_size': '10m',
                                                    'ni_per': 50000},
                                 'collection_params': {'shards_num': 1,
                                                       'collection_name': 'lb_replica3'},
                                 'load_params': {'replica_number': 3},
                                 'index_params': {'index_type': 'HNSW',
                                                  'index_param': {'M': 8,
                                                                  'efConstruction': 200}},
                                 'concurrent_params': {'concurrent_number': 300,
                                                       'during_time': '3h',
                                                       'interval': 20,
                                                       'spawn_rate': None},
                                 'concurrent_tasks': [{'type': 'search',
                                                       'weight': 1,
                                                       'params': {'nq': 10,
                                                                  'top_k': 10,
                                                                  'search_param': {'ef': 64},
                                                                  'timeout': 3600}}]},
            'run_id': 2023083079367431,
            'datetime': '2023-08-30 12:18:56.627376',
            'client_version': '2.2'},
 'result': {'test_result': {'index': {'RT': 205.5679},
                            'insert': {'total_time': 339.616,
                                       'VPS': 29445.0203,
                                       'batch_time': 1.6981,
                                       'batch': 50000},
                            'flush': {'RT': 2.5328},
                            'load': {'RT': 14.6443},
                            'Locust': {'Aggregated': {'Requests': 4031531,
                                                      'Fails': 12129,
                                                      'RPS': 373.29,
                                                      'fail_s': 0.0,
                                                      'RT_max': 27616.26,
                                                      'RT_avg': 791.23,
                                                      'TP50': 740.0,
                                                      'TP99': 1200.0},
                                       'search': {'Requests': 4031531,
                                                  'Fails': 12129,
                                                  'RPS': 373.29,
                                                  'fail_s': 0.0,
                                                  'RT_max': 27616.26,
                                                  'RT_avg': 791.23,
                                                  'TP50': 740.0,
                                                  'TP99': 1200.0}}}}}
```
<img width="1858" alt="截屏2023-08-31 11 05 22" src="https://github.com/milvus-io/milvus/assets/26307815/94005e87-3a9f-4066-b3bd-77e56070488a">
